stream_alert/alert_processor/outputs.py

```diff
@@ -512,20 +513,33 @@ def dispatch(self, **kwargs):
                 alert [dict]: Alert relevant to the triggered rule
         """
         alert = kwargs['alert']
-        service = alert['metadata']['source']['service']
-        entity = alert['metadata']['source']['entity']
+        service = alert['source_service']
+        entity = alert['source_entity']
+
         current_date = datetime.now()
-        alert_string = json.dumps(alert)
+
+        s3_alert = alert
+        # JSON dump the alert to retain a consistent alerts schema across log types.
+        # This will get replaced by a UUID which references a record in a 
+        # different table in the future.
+        s3_alert['record'] = json.dumps(s3_alert['record'])
+        alert_string = json.dumps(s3_alert)
+
         bucket = self.config[self.__service__][kwargs['descriptor']]
-        key = '{}/{}/{}/dt={}/streamalerts_{}.json'.format(
+
+        # Prefix with alerts to account for generic non-streamalert buckets
+        # Produces the following key format:
+        #   alerts/dt=2017-01-25-00-15/kinesis_mystream_myrule_<uuid>.json
+        key = 'alerts/dt={}/{}_{}_{}_{}.json'.format(
+            current_date.strftime('%Y-%m-%d-%H-%M'),
             service,
             entity,
-            kwargs['rule_name'],
-            current_date.strftime('%Y-%m-%d-%H-%M'),
-            current_date.isoformat('-')
+            alert['rule_name'],
+            # keys need to be unique to avoid object overwriting
```

Comment:
Can this comment just get lumped in with the ones above. Placing it in the `format` params makes it look like a param and confusing/odd.